EIPS/eip-remove_utf8_support.md

```diff
@@ -0,0 +1,140 @@
+## Preamble
+
+EIP: -
+
+Title: Remove UTF-8 support from contract strings
+
+Author: Andreas Olofsson (androlo1980@gmail.com)
+
+Type: Standard Track
+
+Category: ERC
+
+Status: Draft
+
+Created: 2017-10-16
+
+## Simple Summary
+
+This proposal is to remove UTF-8 support from contracts, and instead use a simpler and more generic string format.
+
+## Abstract
+
+Strings used in contracts are currently assumed to be UTF-8 encoded. There are no enforcement mechanisms, however, which means that a `string` is essentially the same as a `bytes`. An alternative to this is to define stings to be a more general form of `bytes`, which can be done by changing the the format of the header. The strings proposed here would allow user-defined alphabets with `O(1)` index access and length computation. Additionally, runtime validation of any given string would be done using a very simple compare function.
+
+## Motivation
+
+UTF-8 is not a good string format for contracts, for several reasons:
+
+1. It uses variable length characters, which means you have to index the strings before using them, or iterate over parts of the string when you want to get the length, or get a character by index.
```

Comment:
Getting a character by index is rarely necessary with a well-written string library, however; operations like partitioning and splitting are far more common.

Reply:
maybe get is the wrong word. to be more specific: "get the position of a character in the backing byte array, by providing the index of the character in the string".